tests/test_converter.py

```diff
@@ -0,0 +1,1221 @@
+        ):
+            load_program(program)(wires=range(3))
+
+
+class TestQuilConverter:
+    """Test that quil instances passed as string are properly converted."""
+
+    def test_convert_simple_program(self):
+        """Test that a simple program is properly converted."""
+        quil_str = textwrap.dedent(
+            """
+            H 0
+            RZ(0.34) 1
+            CNOT 0 3
+            H 2
+            H 7
+            X 7
+            Y 1
+            RZ(0.34) 1
+        """
+        )
+
+        with OperationRecorder() as rec:
+            load_quil(quil_str)(wires=range(5))
+
+        # The wires should be assigned as
+        # 0  1  2  3  7
+        # 0  1  2  3  4
+
+        expected_queue = [
+            qml.Hadamard(0),
+            qml.RZ(0.34, wires=[1]),
+            qml.CNOT(wires=[0, 3]),
+            qml.Hadamard(2),
+            qml.Hadamard(4),
+            qml.PauliX(4),
+            qml.PauliY(1),
+            qml.RZ(0.34, wires=[1]),
+        ]
+
+        for converted, expected in zip(rec.queue, expected_queue):
+            assert converted.name == expected.name
+            assert converted.wires == expected.wires
+            assert converted.params == expected.params
+
+    def test_convert_program_with_classical_control_flow(self):
+        """Test that a program with classical control flow is properly converted."""
+        quil_str = textwrap.dedent(
+            """
+            DECLARE flag_register BIT[1]
+            MOVE flag_register 1
+            LABEL @START1
+            JUMP-UNLESS @END2 flag_register
+            X 0
+            H 0
+            MEASURE 0 flag_register
+            JUMP @START1
+            LABEL @END2
+        """
+        )
+
+        flag = 0.0
+
+        with OperationRecorder() as rec:
+            load_quil(quil_str)(wires=[0], parameter_map={"flag_register": flag})
+
+        expected_queue = [
+            qml.PauliX(0),
+            qml.Hadamard(0),
+        ]
+
+        for converted, expected in zip(rec.queue, expected_queue):
+            assert converted.name == expected.name
+            assert converted.wires == expected.wires
+            assert converted.params == expected.params
+
+    def test_convert_program_with_pragmas(self):
+        """Test that a program with pragmas is properly converted."""
+        quil_str = textwrap.dedent(
+            """
+            PRAGMA INITIAL_REWIRING "GREEDY"
+            PRAGMA PRESERVE_BLOCK
+            I 0
+            I 1
+            PRAGMA END_PRESERVE_BLOCK
+            PRAGMA COMMUTING_BLOCKS
+            PRAGMA BLOCK
+            CNOT 0 1
+            RZ(0.4) 1
+            CNOT 0 1
+            PRAGMA END_BLOCK
+            PRAGMA BLOCK
+            CNOT 1 2
+            RZ(0.6) 2
+            CNOT 1 2
+            PRAGMA END_BLOCK
+            PRAGMA END_COMMUTING_BLOCKS
+            H 0
+        """
+        )
```

Comment:
Is this just testing that pragmas are ignored, as intended?

Reply:
Exactly.